services-turf/src/main/java/com/mapbox/turf/TurfConversion.java

```diff
@@ -200,5 +206,63 @@ public static FeatureCollection explode(@NonNull Feature feature) {
       finalFeatureList.add(Feature.fromGeometry(singlePoint));
     }
     return FeatureCollection.fromFeatures(finalFeatureList);
+
+  }
+
+  /**
+   * Combines a FeatureCollection of all {@link Point}s , {@link LineString}s,
+   * or {@link Polygon} features into {@link MultiPoint}, {@link MultiLineString},
+   * or {@link MultiPolygon}.
+   *
+   * @param featureCollection a {@link FeatureCollection} that has {@link Feature}
+   *                          objects which are all {@link Point}, {@link LineString},
+   *                          or {@link Polygon} geometries.
+   * @return a {@link FeatureCollection} with a multi geometry.
+   * @since 4.8.0
+   */
+  public static FeatureCollection combine(@NonNull FeatureCollection featureCollection) {
+    Geometry firstFeatureGeometry = featureCollection.features().get(0).geometry();
+    if (firstFeatureGeometry instanceof Polygon) {
+      List<Polygon> polygonList = new ArrayList<>();
+      for (Feature singleFeature : featureCollection.features()) {
+        if (singleFeature.geometry() instanceof Polygon) {
+          polygonList.add((Polygon) singleFeature.geometry());
+        } else {
+          throw new TurfException("Your FeatureCollection must be of all of "
+            + "the same geometry type.");
+        }
+      }
+      return FeatureCollection.fromFeature(Feature.fromGeometry(
+        MultiPolygon.fromPolygons(polygonList)));
+    } else if (firstFeatureGeometry instanceof Point) {
+      List<Point> pointList = new ArrayList<>();
+      for (Feature singleFeature : featureCollection.features()) {
+        if (singleFeature.geometry() instanceof Point) {
+          pointList.add((Point) singleFeature.geometry());
+        } else {
+          throw new TurfException("Your FeatureCollection must be of all of "
+            + "the same geometry type.");
+        }
+      }
+      return FeatureCollection.fromFeature(Feature.fromGeometry(
+        MultiPoint.fromLngLats(pointList)));
+    } else if (firstFeatureGeometry instanceof LineString) {
+      List<LineString> lineStringList = new ArrayList<>();
+      for (Feature singleFeature : featureCollection.features()) {
+        if (singleFeature.geometry() instanceof LineString) {
+          lineStringList.add((LineString) singleFeature.geometry());
+        } else {
+          throw new TurfException("Your FeatureCollection must be of all of "
+            + "the same geometry type.");
+        }
+      }
+      return FeatureCollection.fromFeature(Feature.fromGeometry(
+        MultiLineString.fromLineStrings(lineStringList)
+      ));
+    }
+    throw new TurfException("Your FeatureCollection did not include "
+      + "a Point, LineString, or Polygon geometry type. Please pass in "
+      + "a FeatureCollection with two or more Features which are all of "
+      + "the same geometry type.");
```

Comment:
Above block doesn't match the js one in https://github.com/Turfjs/turf/blob/master/packages/turf-combine/index.ts. 

> Support multiple geometry types

We need to be able to support multiple types of geometry in one resulting feature collection. For example if you would combine FeatureCollection of Points and FeatureCollection of LineString in your input. You would get a FeatureCollection with of 2 features (1 multipoint and 1 multilinestring) as output. 


> Support all input

We are currently taking the 0 index of the input, we need to use all input.  

```
 featureCollection.features().get(0).geometry();
```

Reply:
> We need to be able to support multiple types of geometry in one resulting feature collection

Addressed in my latest commit. 

> For example if you would combine FeatureCollection of Points and FeatureCollection of LineString in your input. You would get a FeatureCollection with of 2 features (1 multipoint and 1 multilinestring) as output

My understanding of this method is a little different than ⬆️ ; based on http://turfjs.org/docs/#combine and the test lines such as https://github.com/Turfjs/turf/blob/master/packages/turf-combine/test.js#L24 and https://github.com/Turfjs/turf/blob/master/packages/turf-combine/test.js#L142.

A single `FeatureCollection` is the parameter. The `FeatureCollection` has to have `Point` and/or `MultiPoint`, `LineString` and/or `MultiLineString`, or `Polygon` and/or `MultiPolygon`. No mixing `Point`-based geometries with `Polygon`-based, etc.

The returned `FeatureCollection` only has _one_ `Feature` in it and the single `Feature` is always of a "Multi-" geometry.

I basically explain this at https://github.com/mapbox/mapbox-java/blob/b3b20132ec24b223c12aaac42e96e88332f72d2e/services-turf/src/main/java/com/mapbox/turf/TurfConversion.java#L213-L226

What's a little weird is that http://turfjs.org/docs/#combine shows that non-Multi geometries are the only types that can be in the parameter `FeatureCollection`. Whereas, the tests allow for "Multi-" geometries (https://github.com/Turfjs/turf/blob/master/packages/turf-combine/test.js#L142 and https://github.com/Turfjs/turf/blob/master/packages/turf-combine/test.js#L69). My latest commit follows the tests so that "Multi-" geometries are supported (e.g. `Point` and `MultiPoint` are in the parameter `FeatureCollection`).


> We are currently taking the 0 index of the input, we need to use all input.

Addressed in my latest commit.